### PR TITLE
Code for automatically updating license headers

### PR DESCRIPTION
**Pull request by stes**
## More updates:

- As discussed, reverted the actual changes to the license headers for now
- Added a new `tools` directory for future developer tools (like pre-commit functions, etc.)

## Some updates:

It is not possible to configure various license statements directly in `NOTICE.yml`. This is also useful in general to have an overview over third partly licenses applicable to the code.

For updating license headers in the whole repository, simply run

```
python update_license_headers.py
```

now. Some TODOs:

- [x] **Before merging**, we need to remove the old docstring license headers. I left them in this PR to more easily compare if the correct licenses was applied everywhere. I can do that once I got that approval.
- [x] It should be decided where to put the `update_license_headers.py` helper script. Afaik there is not "developer tools" directory right now. The options I can think of are: (1) leave it in the repo root (2) put it in `.github` or (3) add it to `deeplabcut.utils`. Opinions?

## Original PR description

Prior to this PR, the license headers in the codebase had two issues:

- They were not standardized, and the version information was outdated ("DeepLabCut2.0", etc.)
- They were added to the modules `__doc__` variable because
- Potentially maintainers used some automated tool for this, but the correct license header to use was not added to the repo

This PR proposes the following changes:
- Use license headers that are started by `#` instead of tripe quotes (`"""`). This avoids a clash between the module docstring and makes it easier to write API docs on module level.
- Update the contribution guide with instructions of how to automatically update the license headers (see below)
- Add license headers to all files in `tests/` and `examples/`
- Updated the header to ~"DeepLabCut2.0-2.3 Toolbox" (the version info could optionally also be automatically inserted, if desired)~ DeepLabCut Toolbox
- License header can now be configured in `.copyright.tmpl` file

The headers can always be updated by running (this can be automated and e.g. be added to a script that also triggers code formatting w/ black, etc):

```bash
find deeplabcut tests examples -iname '*.py' -type f \
  -exec licenseheaders -t .copyright.tmpl -f {} \;
licenseheaders -t .copyright.tmpl -f dlc.py
```

TODO (for maintainers):
- [x] Please check if the info in `.copyright.tmpl` is accurate/should be updated
- [x] update non-dlc copy right headers accordingly (see review comments)

### CODE REVIEW

File: .copyright.tmpl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+DeepLabCut2.0-2.3 Toolbox (deeplabcut.org)
```
Comment (stes) [inline .copyright.tmpl:1]:
```suggestion
DeepLabCut Toolbox (deeplabcut.org)
```

**Review by MMathisLab — APPROVED** [review]

File: deeplabcut/pose_estimation_tensorflow/backbones/efficientnet_builder.py [hunk 2]
```diff
@@ -1,17 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
 #
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/backbones/efficientnet_builder.py]: header incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/config.py [hunk 3]
```diff
@@ -1,3 +1,13 @@
+#
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/config.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_deterministic.py [hunk 4]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_deterministic.py]: remove

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_imgaug.py [hunk 5]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_imgaug.py]: remove

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_scalecrop.py [hunk 6]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_scalecrop.py]: remove

File: deeplabcut/pose_estimation_tensorflow/datasets/pose_tensorpack.py [hunk 7]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/datasets/pose_tensorpack.py]: remove

File: deeplabcut/pose_estimation_tensorflow/default_config.py [hunk 8]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/default_config.py]: remove

File: deeplabcut/pose_estimation_tensorflow/lib/__init__.py [hunk 9]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/lib/__init__.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/__init__.py [hunk 10]
```diff
@@ -1,3 +1,13 @@
...
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/__init__.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/conv_blocks.py [hunk 11]
```diff
@@ -1,17 +1,13 @@
-# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/conv_blocks.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/nnets/efficientnet.py [hunk 12]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut 2.1.9 Toolbox (deeplabcut.org)
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/efficientnet.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/mobilenet.py [hunk 13]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.2 Toolbox (deeplabcut.org)
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/mobilenet.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/multi.py [hunk 14]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.2 Toolbox (deeplabcut.org)
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/multi.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/resnet.py [hunk 15]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.2 Toolbox (deeplabcut.org)
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/resnet.py]: remove

File: deeplabcut/pose_estimation_tensorflow/nnets/utils.py [hunk 16]
```diff
@@ -1,17 +1,13 @@
-# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# DeepLabCut Toolbox (deeplabcut.org)
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/nnets/utils.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/util/__init__.py [hunk 17]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Adapted from DeeperCut by Eldar Insafutdinov:
 https://github.com/eldar/pose-tensorflow
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/util/__init__.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/util/logging.py [hunk 18]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Adapted from DeeperCut by Eldar Insafutdinov
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/util/logging.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/util/visualize.py [hunk 19]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Adapted from DeeperCut by Eldar Insafutdinov
 https://github.com/eldar/pose-tensorflow
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/util/visualize.py]: incorrectly updated

File: deeplabcut/pose_estimation_tensorflow/vis_dataset.py [hunk 20]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Adapted from DeeperCut by Eldar Insafutdinov
 https://github.com/eldar/pose-tensorflow
```
Comment (stes) [inline deeplabcut/pose_estimation_tensorflow/vis_dataset.py]: incorrectly updated

File: deeplabcut/pose_tracking_pytorch/solver/scheduler_factory.py [hunk 21]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """ Scheduler Factory
 Hacked together by / Copyright 2020 Ross Wightman
 """
```
Comment (stes) [inline deeplabcut/pose_tracking_pytorch/solver/scheduler_factory.py]: Incorrectly updated

File: deeplabcut/pose_tracking_pytorch/tracking_utils/reranking.py [hunk 22]
```diff
@@ -1,5 +1,15 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Created on Fri, 25 May 2018 20:29:09
```
Comment (stes) [inline deeplabcut/pose_tracking_pytorch/tracking_utils/reranking.py]: is this an original DLC file or taken from the repo?

File: deeplabcut/post_processing/analyze_skeleton.py [hunk 23]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.0 Toolbox (deeplabcut.org)
 © A. & M. Mathis Labs
```
Comment (stes) [inline deeplabcut/post_processing/analyze_skeleton.py]: not removed

File: deeplabcut/utils/make_labeled_video.py [hunk 24]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.0 Toolbox (deeplabcut.org)
 © A. & M. Mathis Labs
```
Comment (stes) [inline deeplabcut/utils/make_labeled_video.py]: not removed

File: deeplabcut/utils/video_processor.py [hunk 25]
```diff
@@ -1,3 +1,13 @@
...
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 Author: Hao Wu
 hwu01@g.harvard.edu
```
Comment (stes) [inline deeplabcut/utils/video_processor.py]: incorrectly updated

File: examples/testscript_3d.py [hunk 26]
```diff
@@ -1,3 +1,13 @@
...
+# © A. & M. Mathis Labs
+# https://github.com/DeepLabCut/DeepLabCut
+#
+# Please see AUTHORS for contributors.
+# https://github.com/DeepLabCut/DeepLabCut/blob/master/AUTHORS
+#
+# Licensed under GNU Lesser General Public License v3.0
+#
 """
 DeepLabCut2.0 Toolbox (deeplabcut.org)
```
Comment (stes) [inline examples/testscript_3d.py]: not removed

File: examples/testscript_openfielddata.py [hunk 27]
```diff
@@ -1,5 +1,15 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
```
Comment (stes) [inline examples/testscript_openfielddata.py]: add these to other testscripts?

**Review by stes — COMMENTED** [review]
identified some incorrectly updated headers

**Comment by stes** [comment]
@AlexEMG pls see my changes:

It is not possible to configure various license statements directly in `NOTICE.yml`. This is also useful in general to have an overview over third partly licenses applicable to the code.

For updating license headers in the whole repository, simply run

```
python update_license_headers.py
```

now. Some TODOs:

- [ ] **Before merging**, we need to remove the old docstring license headers. I left them in this PR to more easily compare if the correct licenses was applied everywhere. I can do that once I got that approval.
- [ ] It should be decided where to put the `update_license_headers.py` helper script. Afaik there is not "developer tools" directory right now. The options I can think of are: (1) leave it in the repo root (2) put it in `.github` or (3) add it to `deeplabcut.utils`. Opinions?

**Comment by MMathisLab** [comment]
"It should be decided where to put the update_license_headers.py helper script. Afaik there is not "developer tools" directory right now. The options I can think of are: (1) leave it in the repo root (2) put it in .github or (3) add it to deeplabcut.utils. Opinions?"

- we could make a dev tooling dir?

**Review by AlexEMG — APPROVED** [review]

**Comment by stes** [comment]
## More updates:

- As discussed, reverted the actual changes to the license headers for now
- Added a new `tools` directory for future developer tools (like pre-commit functions, etc.)

@AlexEMG @MMathisLab

**Review by AlexEMG — APPROVED** [review]
Thanks, @stes!

**Pull request merged**